protostar/cli/config_file_argument_resolver.py

```diff
@@ -0,0 +1,8 @@
+from typing import Any, Optional, Protocol
+
+
+class ConfigFileArgumentResolverProtocol(Protocol):
```

Comment:
I discussed it with @Arcticae. We shouldn't generalize abstract interfaces when only one class implements it. Point of abstraciton is to generalize something and we should be aware of complexity it introduces. 

Please:
- remove this protocol
- use mock of the exact class in tests instead

Reply:
Please provide an alternative solution that follows [acyclic dependencies principle](https://en.wikipedia.org/wiki/Acyclic_dependencies_principle).

Reply:
Can you point the cycle proposed solution creates?

Reply:
https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)

Reply:
@kasperski95 is there a cycle when trying to mock the said class?

Reply:
> Can you point the cycle proposed solution creates?

> @kasperski95 is there a cycle when trying to mock the said class?

I don't know the exact cycle (they are hard to reason about), but I know it exists because:
- it breaks Protostar and VS Code testing extension.
- `protostar.cli` imports from `protostar.configuration_file` and `protostar.configuration_file` imports from `protostar.cli`

> https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming), on the other hand your solution breaks other rule.

The point of that interface is not creating abstraction but creating boundaries, hence the rule of three doesn't really apply here.

> In essence, a system’s architecture is what defines the shape of the system. More specifically, a system’s architecture defines how the system is divided into components, how those components are arranged, what of kinds boundaries exist between different components and how the components communicate across those boundaries. Basically, it’s all about the way we are using boundaries to separate parts of the system that shouldn’t know too much about each other.
https://convincedcoder.com/2019/04/27/Software-architecture-boundaries/
 
> use mock of the exact class in tests instead

I can use them instead of stubs and use the rule of three here, (maybe in the future PRs). Mocks create more fragile tests and pollute test cases with implementation details.

> I think it would be easier and simpler to create a new class than to isolate an abstract protocol.

A developer would be responsible for keeping that new class and `ConfigurationFile` concrete class in sync.